src/overlays/actors/ovl_En_Anubice/z_en_anubice.c

```diff
@@ -53,35 +63,441 @@ static ColliderCylinderInit D_809B22D0 = {
+    f32 zero;
+    BgHidanCurtain* hidanCurtain;
+    s32 i;
+    Vec3f sp48;
+    Vec3f sp3C;
+    EnAnubice* this = THIS;
+
+    if (this->actionFunc != EnAnubice_SetupDie && this->actionFunc != EnAnubice_Die &&
+        this->actor.shape.yOffset == 0.0f) {
+        EnAnubice_Hover(this, globalCtx);
+        for (i = 0; i < 5; i++) {
+            hidanCurtain = this->hidanCurtains[i];
+
+            if (hidanCurtain != NULL && fabsf(hidanCurtain->actor.world.pos.x - this->actor.world.pos.x) < 60.0f &&
+                fabsf(this->hidanCurtains[i]->actor.world.pos.z - this->actor.world.pos.z) < 60.0f &&
+                hidanCurtain->timer != 0) {
+                Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORCAT_PROP);
+                this->actor.flags &= ~1;
+                func_80032C7C(globalCtx, &this->actor);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_ANUBIS_DEAD);
+                this->actionFunc = EnAnubice_SetupDie;
+                return;
+            }
+        }
+
+        if (this->col.base.acFlags & 2) {
+            this->col.base.acFlags &= ~2;
+            if (this->actor.colChkInfo.damageEffect == 2) {
+                Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORCAT_PROP);
+                this->actor.flags &= ~1;
+                func_80032C7C(globalCtx, &this->actor);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_ANUBIS_DEAD);
+                this->actionFunc = EnAnubice_SetupDie;
+                return;
+            }
+
+            if (!this->isKnockedback) {
+                this->knockbackTimer = 10;
+                this->isKnockedback = 1;
+
+                sp48.x = 0.0f;
+                sp48.y = 0.0f;
+                sp48.z = -10.0f;
+                sp3C.x = 0.0f;
+                sp3C.y = 0.0f;
+                sp3C.z = 0.0f;
+
+                Matrix_RotateY((this->actor.shape.rot.y / 32768.0f) * M_PI, MTXMODE_NEW);
+                Matrix_MultVec3f(&sp48, &sp3C);
+
+                this->actor.velocity.x = sp3C.x;
+                this->actor.velocity.z = sp3C.z;
+                this->knockbackRecoveryVelocity.x = -sp3C.x;
+                this->knockbackRecoveryVelocity.z = -sp3C.z;
+
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_CUTBODY);
+            }
+        }
+
+        if (this->isKnockedback) {
+            this->actor.shape.rot.y += 6500;
+            Math_ApproachF(&this->actor.velocity.x, this->knockbackRecoveryVelocity.x, 0.3f, 1.0f);
+            Math_ApproachF(&this->actor.velocity.z, this->knockbackRecoveryVelocity.z, 0.3f, 1.0f);
+
+            zero = 0.0f;
+            if (zero) {}
+
+            if (this->knockbackTimer == 0) {
+                this->actor.velocity.z = zero;
+                this->actor.velocity.x = zero;
+                this->knockbackRecoveryVelocity.z = 0.0f;
+                this->knockbackRecoveryVelocity.x = 0.0f;
```

Comment:
These look like chained assigns. Also might help with the `zero` thing?

Reply:
I determined that `zero` does not actually need to be assigned to any of these variables, but it seems to still need to exist. Zero needs to be loaded into the register before `if (this->knockbackTimer == 0)` is checked.